features/netsocket/NetworkStack.h

```diff
@@ -63,13 +62,56 @@ class NetworkStack
     virtual nsapi_error_t gethostbyname(const char *host,
             SocketAddress *address, nsapi_version_t version = NSAPI_UNSPEC);
 
+    /** Hostname translation callback (asynchronous)
+     *
+     *  Callback will be called after DNS resolution completes or a failure
+     *  occurs.
+     *
+     *  @param status  0 on success, negative error code on failure
+     *  @param address On success, destination for the host SocketAddress
+     *  @param data    Caller defined data
+     */
+    typedef mbed::Callback<void (nsapi_error_t result, SocketAddress *address, void *data)> hostbyname_cb_t;
```

Comment:
If the query returns multiple addresses, is there a way to iterate through all of them, just as with getaddrinfo() on POSIX?

Reply:
Like previous blocking interface, multiple addresses are not supported. That might be done later as a new extension to interface.